### DISCUSSION TITLE
Cost of embedding in langfuse

### DISCUSSION THREAD

**Discussion by Amitt1412** [langfuse/langfuse#2486] [Support] [answered]
Hi folks, I tried logging the cost of embedding through this following function 
# Capture OpenAI usage and cost
        if 'usage' in response:
            embedding_cost = resources.retrieval_embedding_model.get_embedding_cost()
            langfuse.generation(
                model=resources.retrieval_embedding_model,
                input=[{"role": "user", "text": query}],
                output=response['text'],
                usage={
                    "prompt_tokens": response['usage']['prompt_tokens'],
                    "completion_tokens": response['usage']['completion_tokens'],
                    "total_tokens": response['usage']['total_tokens'],
                    "Total_Embedding_Cost": embedding_cost,  # Log the embedding cost
                    "embedding_model": resources.retrieval_embedding_model, 
                    "embedding_cost": embedding_cost,  
                }
            )
        return response

but throwing the error as no attribute - get_embedding_cost(). How can I get it corrected? Any idea anyone can suggest?

**Comment by marcklingen** [comment]
Hi, is `get_embedding_cost` mentioned somehwere in the langfuse docs? I could not find it.

The usage object does not take these keys:
```
"Total_Embedding_Cost": embedding_cost,  # Log the embedding cost
"embedding_model": resources.retrieval_embedding_model, 
"embedding_cost": embedding_cost,  
```

Use the standard usage object mentioned here: https://langfuse.com/docs/model-usage-and-cost#ingest

**Comment by Amitt1412** [comment]
Hi marklingen,

Actully it was through the following snippet later on 
chain = qa_template | resources.llm 
with get_openai_callback() as cb:
    result = resources.llm.invoke(query, config={"callbacks": [langfuse_handler]})
    print(cb)
langfuse_handler.metadata["Embedding_Tokens"] = cb.total_tokens
langfuse_handler.metadata["Embedding_cost"] = cb.total_cost

**Reply by marcklingen** [reply, marked as answer]
Thanks for sharing, you cannot set tokens on the attributes of the langfuse_handler like in your snippet unfortunately, you'd need to log them via the low-level Python SDK as a separate generation object
